libcontainer/configs/validate/validator.go

```diff
@@ -253,16 +253,72 @@ func cgroupsCheck(config *configs.Config) error {
 	return nil
 }
 
+func checkIDMapMounts(config *configs.Config, m *configs.Mount) error {
+	if !m.IsIDMapped() {
+		return nil
+	}
+
+	if !m.IsBind() {
+		return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is supported only for mounts with the option 'bind'", m)
+	}
+	if config.RootlessEUID {
+		return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is not supported when runc is being launched with EUID != 0, needs CAP_SYS_ADMIN on the runc parent's user namespace", m)
```

Comment:
We talked about this same thing with Kir here: https://github.com/opencontainers/runc/pull/3717#discussion_r1154707177

As I mentioned there, I understand this _can_ be too long, but it is also _super_ rare to hit. And whenever we hit it, is not that we can ask the user to see the config.json easily: k8s deletes it and retries super quick, it is not easy to find on which folder it is, etc.

And this is only hit in the case of users willing opting in for the feature.

I think having all the info about the mount, to understand what exactly we are dealing with, and maybe the info there can help us to see where in the code of other projects this config.json mount is being generated.


Did I convince you too? :)

Reply:
> As I mentioned there, I understand this can be too long, but it is also super rare to hit. And whenever we hit it, is not that we can ask the user to see the config.json easily: k8s deletes it and retries super quick, it is not easy to find on which folder it is, etc.

Yup, not a blocker for me, and now that that part of the error is in a single location (where we wrap the error), it will be easier to update the formatting if we would decide to 👍